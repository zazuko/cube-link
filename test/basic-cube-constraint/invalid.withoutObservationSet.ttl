@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix dc: <http://purl.org/dc/terms/> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix sh: <http://www.w3.org/ns/shacl#> .
@prefix cube: <https://cube.link/> .
@base <https://example.org/> .

<cube> a cube:Cube ;
    #cube:observationSet <observationSet> ;
    cube:observationConstraint <constraint> .
<observationSet> cube:observation <observation> .
<constraint> sh:property
  [ sh:path rdf:type ] ,
  [ sh:path cube:observedBy ] ,
  [ sh:path dc:date ] .
