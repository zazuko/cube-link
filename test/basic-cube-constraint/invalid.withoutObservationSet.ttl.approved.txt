@prefix sh: <http://www.w3.org/ns/shacl#> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix cube: <https://cube.link/> .

_:report a sh:ValidationReport ;
	sh:result [
		rdf:type sh:ValidationResult ;
		sh:resultSeverity sh:Violation ;
		sh:sourceConstraintComponent sh:MinCountConstraintComponent ;
		sh:sourceShape [
			sh:path cube:observationSet ;
			sh:minCount 1 ;
			sh:node <https://cube.link/shape/ObservationSetShape> ;
			sh:message "cube:Cube needs at least one cube:ObservationSet" ;
		] ;
		sh:focusNode <https://example.org/cube> ;
		sh:resultPath cube:observationSet ;
		sh:resultMessage "cube:Cube needs at least one cube:ObservationSet" ;
	] ;
	sh:conforms false .
